### PR TITLE
Add rciam metrics role

### PR DESCRIPTION
**Pull request by lionick**
Remember to squash before merge this PR.

### CODE REVIEW

File: roles/rciam-metrics/templates/fastapi/config.py.j2 [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+db = {{ metrics_db }}
+db_admin = {{ metrics_db_admin }}
+db_password = {{ metrics_db_password }}
+database_url = postgresql+psycopg2://{{ metrics_db_admin }}:{{ metrics_db_password }}@{{ metrics_db_hostname }}/{{ metrics_db }}
+
+[oidc_client]
+client_id = {{ oidc_client_id }}
+client_secret = {{ oidc_client_secret }}
+issuer = {{ oidc_client_issuer }}
+openid_connect_url = {{ oidc_client_connect_url }}
```
Comment (NicolasLiampotis) [inline roles/rciam-metrics/templates/fastapi/config.py.j2:11]: This is redundant. The OP configuration MUST be available under issuer + '/.well-known/blahblah'

File: roles/rciam-metrics/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+metrics_path: /srv/rciam-metrics-client/rciam-metrics
+metrics_repo: https://github.com/lionick/rciam-metrics-dev
+metrics_latest_release: rc-example
+metrics_version: main
+metrics_fast_api_folder_name: app
+metrics_api_url: http://localhost:8004/
+metrics_db: metrics-dev
+metrics_db_admin: rciam
+metrics_db_password: secret
+metrics_db_hostname: db.example.com
```
Comment (NicolasLiampotis) [inline roles/rciam-metrics/defaults/main.yml:14]: I'd suggest changing this to:
```
metrics_db:
  host: db.rciam.example.org
  name: metrics-dev
  admin: rciam
  password: secret
```

File: roles/rciam-metrics/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#  - target: "/path/to/target"
+#    link: "/path/to/symlink"
+
+# OIDC Client
+oidc_client_id: client-id
+oidc_client_secret: client-secret
+oidc_client_issuer: client-issuer
+oidc_client_connect_url: client-connect-url
+oidc_client_idtoken_model: client-idtoken-model
+oidc_client_redirect_uri: client-redirect-uri
```
Comment (NicolasLiampotis) [inline roles/rciam-metrics/defaults/main.yml:31]: Each tenant needs to have its own OIDC configuration so I'd suggest changing this to:
```
metrics_oidc:
  tenant1:
    client_id: client-id
    client_secret: client-secret
    client_redirect_uri: https://metrics.rciam.example.org/callback
    issuer: https://tenant1.rciam.example.org/auth/realms/rciam
  #tenant2:
  #  client_id: client-id
  #  client_secret: client-secret
  #  client_redirect_uri: https://metrics.rciam.example.org/callback
  #  issuer: https://tenant2.rciam.example.org/auth/realms/rciam
```

File: roles/rciam-metrics/templates/fastapi/config.py.j2 [hunk 4]
```diff
@@ -0,0 +1,13 @@
+[database_parameters]
+db = {{ metrics_db }}
```
Comment (NicolasLiampotis) [inline roles/rciam-metrics/templates/fastapi/config.py.j2:2]: metrics_db.name

File: roles/rciam-metrics/templates/fastapi/config.py.j2 [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+db_admin = {{ metrics_db_admin }}
```
Comment (NicolasLiampotis) [inline roles/rciam-metrics/templates/fastapi/config.py.j2:3]: metrics_db.admin

File: roles/rciam-metrics/templates/fastapi/config.py.j2 [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+db_password = {{ metrics_db_password }}
```
Comment (NicolasLiampotis) [inline roles/rciam-metrics/templates/fastapi/config.py.j2:4]: metrics_db.password

File: roles/rciam-metrics/templates/fastapi/config.py.j2 [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+database_url = postgresql+psycopg2://{{ metrics_db_admin }}:{{ metrics_db_password }}@{{ metrics_db_hostname }}/{{ metrics_db }}
```
Comment (NicolasLiampotis) [inline roles/rciam-metrics/templates/fastapi/config.py.j2:5]: `database_url = postgresql+psycopg2://{{ metrics_db.admin }}:{{ metrics_db.password }}@{{ metrics_db.host }}/{{ metrics_db.name }}`

File: roles/rciam-metrics/templates/fastapi/config.py.j2 (hunk 1 above)
Comment (NicolasLiampotis) [inline roles/rciam-metrics/templates/fastapi/config.py.j2:11]: This has be adapted to allow per tenant OIDC config

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
The role needs to be adapted to support per tenant OIDC configuration

**Review by NicolasLiampotis — APPROVED** [review]
Will most probably need to introduce a `metrics_hostname` var and use it for the client_redirect_uri. Other than that LGTM

**Comment by Douwe47** [comment]
World is m'y house

**Pull request merged**